### PR TITLE
swarm: add integration with swarm service

### PR DESCRIPTION
**Pull request by farazdagi**
- the swarm support PR adapted to run on top of Geth 1.5.9 (w/o blocking LES)
- web3.js has been updated
- the binary has been tested on real device (Android), to make sure that we don't have LES blocking, and Swarm doesn't introduce any issues with running Status on mobile
- Swarm HTTP proxy starts on :8500 (by the time we receive `node.started` event, swarm and its HTTP gateway are already running)
- Swarm needs private key, and on node start he gets a randomly generated one. Whenever you login into your account, swarm service is restarted with your PK instead.

### CODE REVIEW

**Comment by farazdagi** [comment]
PR updated. 


**Notes:**
- the Swarm implementation on `go-ethereum` side is alpha at best: it has weird bugs, it can throw panics, corrupt its database, have issues with fetching chunks that a present in database. So, while we definitely should proceed and integrate the Swarm (as it allows us to find bugs, file PRs to fix those, make sure we end up with Swarm that has all the capabilities we need), it should be kept in mind that we are integrating against the alpha quality software
- Swarm is **not lightweight**, both when it comes to sync and space required. There's no light version of Swarm, where we are clients-only and able to get data w/o syncing. And when sync is allowed then we need **400Mb** on device, which is kind of huge for mobile.
- HTTP API of Swarm is far from ideal, which is supported by the fact that `go-ethereum` team refactors it https://github.com/ethereum/go-ethereum/pull/3773 (there are still issues with HTTP API, which I will report/patch after that PR gets accepted)
- Currently Swarm is tested as a separate command/service, it is not being tested as protocol running along-side the others (LES, SHH). Although due to really nice way the Geth node protocols can be stacked most of the problems are avoided, still there are lots of weird things if I try to restart Swarm node for example (starting from losing info about peers in hives, and ending with resources not properly deallocated). As there are not many teams atm which use Swarm as sub-process, we should actively contribute back (I've already filed a PR, and have changes for couple of others - waiting to see when that API refactoring PR gets merged)

**Known Issues:**
- I've tested using the following:
  - tested ENS by trying to http://localhost:8500/bzz:/theswarm.eth/
  - tested BZZ by using:
    - http://localhost:8500/bzz:/aa9dd4d23e105d0a2e62da38544112468372cd5ad038fbdc9874b1f51b8e76f2/
    - http://localhost:8500/bzz:/3361662fafc789ec320b92ea06cc09dad8b5912f529ac758003e80189bb28c9f/read.html#52a0602c7fb5bff7049c0c836821c7fed9503f41259681a4f5c4ca18f8fde901
    - http://localhost:8500/bzz:/04006eab0d81009e95ad47ecc52daeb939479bf94037d5225d688f3c0ae4804a/
- When system starts, you need to give some amount of time for hives to saturate (otherwise you will see "no manifest" issue, or some other errors reported). Please note that "saturate" doesn't mean being fully in sync.
- Moreover, if you see some "manifest not found" error, then even when node gets saturated you will still receive this error (up until you restart the the process - btw, restarting Geth node will not fix issue, as Swarm has long list of issues with restarting itself). So, this is a huge issue - and I will tackle Swarm node restart (both of core service, db sync and HTTP API endpoint) **in a separate PR**, as this will involve lots of work in cooperation with Swarm team (as we obviously need to merge to upstream all the fixes we can produce)
- There's no LSS (by analogy with LES, Lightweight Ethereum Sub-Protocol, we need Lightweight Swarm Sub-protocol).

**Proposed Roadmap:**
- well bugs with data fetching and absence of lightweight option for running Swarm seem to be real blockers. However, I still propose we proceed with testing and merging this PR. Reasons:
  - not many (if any) teams run Swarm as sub-protocol on LES client on mobile, and so we need to *locate* and fix as many issues as possible, as it is not likely they will be fixed by others
  - we will need to do all that work anyway, either gradually or in one sitting, and I am all for gradual changes, as it allows to locate problems sooner rather than later
  - while not stable, you can still use Swarm and produce necessary code on RN app side (once Swarm stabilizes, you'll hardly need to change anything as API stays the same)
- that being said, integrating **alpha quality** component, will in turn make our product less stable, and for this I propose the following:
  - we will allow to configure Swarm support in #122 (I will submit corresponding PR within a session)
  - in that configuration we will be able to turn Swarm support off (that way we can test everything, but still exclude the heavyweight Swarm from the release)
  - (❗️ @jarradh I need input on this from) another configurable option will be to run either local version of Swarm (which is heavy and not mobile-ready), or just rely on a HTTP proxy. Indeed, given that we rely on our cluster for discovery anyway, we can have a proxy that will talk with our cluster and return response on Swarm requests. Then on cluster I will make sure that Swarm is fully synced and ready to serve responses. This can be good from marketing point of view, as we will get both lightweight Swarm fully working within a session or two, and full-fledted Swarm node as sub-process once we merge this PR.
- **To sum up:**
  - let's test and merge this PR (disregarding known limitations), and proceed with gradual improvements of full-fledged Swarm sub-process
  - let's have a lightweight Swarm proxy, so client-side can utilize Swarm immediately, w/o imposing even more on mobile platform (both from CPU/Memory for sync to data space requirements)
  - @rasom and @jarradh please let me know your thoughts

**Comment by tiabc** [comment]
Closing due to hanging too long.
